posix/unix.c

```diff
@@ -997,21 +1018,24 @@ int unix_poll(unsigned socket, short events)
 	else {
 		if (events & (POLLIN | POLLRDNORM | POLLRDBAND)) {
 			proc_lockSet(&s->lock);
-			if (_cbuffer_avail(&s->buffer) > 0)
+			if (_cbuffer_avail(&s->buffer) > 0 || (s->nonblock && (s->connect != NULL))) {
```

Comment:
Just for clarity could you explain why this change is needed?

Reply:
Otherwise it would not be possible to detect pending connections that could be `accept()`-ed, *but* this is unclear and actually does not work if accept is set to blocking but one decides to poll the connections before calling it. Pushing a better change in a second